includes/class-freemius.php

```diff
@@ -2058,6 +2063,48 @@ function _submit_uninstall_reason_action() {
             exit;
         }
 
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since  2.1.4
+         */
+        function cancel_subscription_or_trial_ajax_action() {
+            $this->_logger->entrance();
+
+            $this->check_ajax_referer( 'cancel_subscription_or_trial' );
+
+            $this->cancel_subscription_or_trial( fs_request_get( 'plugin_id', $this->get_id() ) );
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since  2.1.4
+         *
+         * @param number   $plugin_id
+         * @param int|null $blog_id
+         */
+        private function cancel_subscription_or_trial( $plugin_id, $blog_id = null ) {

```

Comment:
@fajardoleo I understand that you've used the `$blog_id` since you've leveraged this method to replace the pieces of logic that cancel the subscription in other places in the code, but while it may make sense there, as a standalone method it doesn't make any sense. Why would a method that cancels a subscription/trial will switch the DB cursor to the network's install main blog ID? and only if the given `$blog_id` is numeric. It just doesn't make any sense (unless I'm missing something). Therefore, remove this part from the method and just add it in the place(s) where it was triggering as an additional code that is executed after the method call.